traits/tests/check_observe_timing.py

```diff
@@ -0,0 +1,721 @@
+        a_property = Property({})
+
+        {}
+        def _get_a_property(self):
+            return self.child.name
+
+""")
+
+
+def scenario4_6(property_args, cached_property):
+    """ Time the cases described by scenario 4 or 6 depending on the
+    cached_property argument, using the given property_args argument for the
+    Property trait defintion.
+
+    Parameters
+    ----------
+    property_args - Str
+        The string defining the argument to be passed in the definition of the
+        Property trait.  e.g. "depends_on='child.name'"
+    cached_property - Str
+        The string that will be used to decorate the getter method of the
+        Property.  Expected to be '' for scenario 4 and '@cached_property' for
+        scenario 6.
+
+    Returns
+    -------
+    tuple
+        A 4-tuple containing the time to construct the HasTraits subclass, the
+        time to instantiate it, the time to reassign child, and the time to
+        reassign child.name
+    """
+    construct_parent_with_property = \
+        PARENT_WITH_PROPERTY_CONSTRUCTION_TEMPLATE.format(
+            property_args,
+            cached_property
+        )
+    construction_time = timeit.timeit(
+        stmt=construct_parent_with_property,
+        setup=CONSTRUCT_PARENT_SETUP,
+        number=N
+    )
+
+    instantiation_time = timeit.timeit(
+        stmt=INSTANTIATE_PARENT,
+        setup=CONSTRUCT_PARENT_SETUP + construct_parent_with_property,
+        number=N
+    )
+
+    reassign_child_time = timeit.timeit(
+        stmt=REASSIGN_CHILD,
+        setup=(
+            CONSTRUCT_PARENT_SETUP
+            + construct_parent_with_property
+            + INSTANTIATE_PARENT
+        ),
+        number=N
+    )
+
+    reassign_child_name_time = timeit.timeit(
+        stmt=REASSIGN_CHILD_NAME,
+        setup=(
+            CONSTRUCT_PARENT_SETUP
+            + construct_parent_with_property
+            + INSTANTIATE_PARENT
+        ),
+        number=N
+    )
+
+    return (
+        construction_time,
+        instantiation_time,
+        reassign_child_time,
+        reassign_child_name_time
+    )
+
+
+def report1():
+    """ Print a readable benchmark report for scenario 1. """
+
+    print(
+        """
+    Scenario 1:
+        Given a subclass of HasTraits with a trait defined, e.g. name = Str(),
+        compare having a method with no decorator to one that is decorated with
+        @on_trait_change("name"), one decorated with @observe("name"), and one
+        with @observe(trait('name')).
+
+        The timing we are interested in:
+        (1) Time to import the module, i.e. the class construction.
+        (2) Time to instantiate the HasTraits object.
+        (3) Time for reassigning the trait to a new value.
+
+    Times are reported as:
+    (#): absolute_time  relative_time_to_control
+        """
+    )
+    benchmark_template = ("(1): {0:.5f} {3:5.2f}  (2): {1:.5f} {4:5.2f}  "
+                          "(3): {2:.5f} {5:5.2f}")
```

Comment:
Nit: It can get misaligned in some situations:
```
control -
(1): 1.12156 1.00  (2): 0.03318 1.00  (3): 0.01429 1.00  (4): 0.00063 1.00
depends_on='child.name' -
(1): 1.14459 1.02  (2): 0.84582 25.49  (3): 0.18172 12.72  (4): 0.00079 1.25
```

For example, changing `{1:5.2f}` to `{1:<10.2f}` will ensure the entry fills up the space for, say 10 characters (I made up the '10' here), that would also help better separating the relative number from the absolute time visually.